web/pdf_save_service.js

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2016 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PDFSaveServiceFactory } from "./app.js";
+import { writeXRef } from "../src/core/writer.js";
```

Comment:
I'm *really* surprised that this even works, because it absolutely shouldn't. Most likely, you end up *indirectly* bundling a significant portion of the worker-thread code into the viewer, which you *absolutely* should not do!

Instead you'll need to split that file into main/worker-thread parts, with the code placed in the `src/display/` respectively the `src/core/` directories as needed. Note how functionality is exported in https://github.com/mozilla/pdf.js/blob/master/src/pdf.js and how the viewer accesses those through the `pdfjs-lib` alias.